subsys/net/lib/nrf_cloud/src/nrf_cloud_agps.c

```diff
@@ -339,22 +340,22 @@ static int send_to_modem(void *data, size_t data_len,
 		agps_print(type, data);
 	}
 
-	/* At this point, GPS driver or app-provided socket is assumed. */
 	if (gps_dev) {
-		return gps_agps_write(gps_dev, type_socket2gps(type), data,
-				      data_len);
-	}
-
-	err = nrf_sendto(fd, data, data_len, 0, &type, sizeof(type));
-	if (err < 0) {
-		LOG_ERR("Failed to send AGPS data to modem, errno: %d", errno);
-		err = -errno;
+		/* GPS driver */
+		return gps_agps_write(gps_dev, type_socket2gps(type), data, data_len);
```

Comment:
perhaps do `err = gps_agps_write`, like the other cases, and have one `return`? 
or update the other cases to `return`?

Reply:
You are right, I'll remove `return` from here and just set `err`. Earlier it was like this because there was a "data sent to modem" print at the end of the function which was only used in case of GNSS socket (GPS driver has it's own prints). Anyway, I decided to remove that because it doesn't add much value and an error is printed anyway from `nrf_cloud_agps_process()` if sending the data to modem fails.